pages/index.js

```diff
@@ -1,42 +1,82 @@
-import ActivityCard from "@/components/ActivityCard";
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import useSWR from "swr";
 import Link from "next/link";
 import styled from "styled-components";
 import PlaceholderLogo from "@/Icons/Placeholder";
 import Navigation from "@/components/Navigation";
 import { theme } from "@/styles";
 import SearchBar from "/components/SearchBar";
+import Filter from "@/components/Filter/Index";
+import ActivityCard from "@/components/ActivityCard";
 
 export default function HomePage() {
   const { data: activities, isLoading } = useSWR("/api/activities");
   const [filteredActivities, setFilteredActivities] = useState([]);
   const [searchTerm, setSearchTerm] = useState("");
+  const [authorFilter, setAuthorFilter] = useState("");
+  const [categoryFilter, setCategoryFilter] = useState("");
+  const [showFilterWindow, setShowFilterWindow] = useState(false);
+
   const handleSearch = (searchTerm) => {
     setSearchTerm(searchTerm);
-    if (!searchTerm) {
-      setFilteredActivities([]);
-      return;
+  };
+
+  const handleFilter = ({ author, category }) => {
+    setAuthorFilter(author);
+    setCategoryFilter(category);
+  };
+
+  useEffect(() => {
+    if (!isLoading && activities) {
+      let filtered = activities;
+
+      if (authorFilter) {
+        filtered = filtered.filter(
+          (activity) =>
+            activity.author.toLowerCase() === authorFilter.toLowerCase()
+        );
+      }
+
+      if (categoryFilter) {
+        filtered = filtered.filter(
+          (activity) =>
+            activity.category.toLowerCase() === categoryFilter.toLowerCase()
+        );
+      }
+
+      setFilteredActivities(filtered);
     }
-    const filtered = activities.filter((activity) =>
-      activity.name.toLowerCase().includes(searchTerm.toLowerCase())
-    );
-    setFilteredActivities(filtered);
+  }, [authorFilter, categoryFilter, activities, isLoading]);
```

Comment:
you don't need this useEffect, but can instead outsource the logic into a function and save its return value in a variable. Like so:

```js
function getFilteredActivities() {
    if (!isLoading && activities) {
      let filtered = activities;

      if (authorFilter) {
        filtered = filtered.filter(
          (activity) =>
            activity.author.toLowerCase() === authorFilter.toLowerCase()
        );
      }

      if (categoryFilter) {
        filtered = filtered.filter(
          (activity) =>
            activity.category.toLowerCase() === categoryFilter.toLowerCase()
        );
      }

      return filtered;
    }
  }

  const filteredActivities = getFilteredActivities();
```